unison.2.51.4~rc3 fails to build using opam

Detected in https://github.com/ocaml/opam-repository/pull/18861
```
#=== ERROR while compiling unison.2.51.4~rc3 ==================================#
# context              2.0.8 | linux/x86_64 | ocaml-base-compiler.4.05.0 | pinned(https://github.com/bcpierce00/unison/archive/refs/tags/v2.51.4_rc3.tar.gz)
# path                 ~/.opam/4.05/.opam-switch/build/unison.2.51.4~rc3
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p unison -j 31
# exit-code            1
# env-file             ~/.opam/log/unison-21485-0a5907.env
# output-file          ~/.opam/log/unison-21485-0a5907.out
### output ###
#     ocamlopt src/linkgtk2.exe (exit 2)
# (cd _build/default && /home/opam/.opam/4.05/bin/ocamlopt.opt -w -40 -w -3-6-9-27-32-52 -g -o src/linkgtk2.exe /home/opam/.opam/4.05/lib/ocaml/unix.cmxa -I /home/opam/.opam/4.05/lib/ocaml /home/opam/.opam/4.05/lib/ocaml/threads/threads.cmxa -I /home/opam/.opam/4.05/lib/ocaml /home/opam/.opam/4.05/lib/ocaml/str.cmxa -I /home/opam/.opam/4.05/lib/ocaml src/lwt/lwt_lib.cmxa /home/opam/.opam/4.05/lib/ocaml/bigarray.cmxa -I /home/opam/.opam/4.05/lib/ocaml src/unison_lib.cmxa -I src /home/opam/.opam/4.05/lib/lablgtk2/lablgtk.cmxa /home/opam/.opam/4.05/lib/lablgtk2/gtkThread.cmx -I /home/opam/.opam/4.05/lib/lablgtk2 src/.linkgtk2.eobjs/native/uigtk2.cmx src/.linkgtk2.eobjs/native/linkgtk2.cmx)
# /usr/bin/ld: src/unison_lib.a(uutil.o): in function `camlUutil__hash_1212':
# /home/opam/.opam/4.05/.opam-switch/build/unison.2.51.4~rc3/_build/default/src/uutil.ml:39: undefined reference to `unsn_hash_univ_param'
# /usr/bin/ld: src/unison_lib.a(uutil.o): in function `camlUutil__43':
# :(.data+0x548): undefined reference to `unsn_hash_univ_param'
# /usr/bin/ld: src/unison_lib.a(case.o): in function `camlCase__fun_1815':
# /home/opam/.opam/4.05/.opam-switch/build/unison.2.51.4~rc3/_build/default/src/uutil.ml:39: undefined reference to `unsn_hash_univ_param'
# /usr/bin/ld: src/unison_lib.a(case.o): in function `camlCase__fun_1836':
# /home/opam/.opam/4.05/.opam-switch/build/unison.2.51.4~rc3/_build/default/src/uutil.ml:39: undefined reference to `unsn_hash_univ_param'
# /usr/bin/ld: src/unison_lib.a(case.o): in function `camlCase__fun_1860':
# /home/opam/.opam/4.05/.opam-switch/build/unison.2.51.4~rc3/_build/default/src/uutil.ml:39: undefined reference to `unsn_hash_univ_param'
# /usr/bin/ld: src/unison_lib.a(case.o):/home/opam/.opam/4.05/.opam-switch/build/unison.2.51.4~rc3/_build/default/src/uutil.ml:39: more undefined references to `unsn_hash_univ_param' follow
# collect2: error: ld returned 1 exit status
# File "caml_startup", line 1:
# Error: Error during linking
#     ocamlopt src/linktext.exe (exit 2)
# (cd _build/default && /home/opam/.opam/4.05/bin/ocamlopt.opt -w -40 -g -o src/linktext.exe /home/opam/.opam/4.05/lib/ocaml/str.cmxa -I /home/opam/.opam/4.05/lib/ocaml /home/opam/.opam/4.05/lib/ocaml/unix.cmxa -I /home/opam/.opam/4.05/lib/ocaml src/lwt/lwt_lib.cmxa /home/opam/.opam/4.05/lib/ocaml/bigarray.cmxa -I /home/opam/.opam/4.05/lib/ocaml src/unison_lib.cmxa -I src src/.linktext.eobjs/native/linktext.cmx)
# /usr/bin/ld: src/unison_lib.a(uutil.o): in function `camlUutil__hash_1212':
# /home/opam/.opam/4.05/.opam-switch/build/unison.2.51.4~rc3/_build/default/src/uutil.ml:39: undefined reference to `unsn_hash_univ_param'
# /usr/bin/ld: src/unison_lib.a(uutil.o): in function `camlUutil__43':
# :(.data+0x548): undefined reference to `unsn_hash_univ_param'
# /usr/bin/ld: src/unison_lib.a(case.o): in function `camlCase__fun_1815':
# /home/opam/.opam/4.05/.opam-switch/build/unison.2.51.4~rc3/_build/default/src/uutil.ml:39: undefined reference to `unsn_hash_univ_param'
# /usr/bin/ld: src/unison_lib.a(case.o): in function `camlCase__fun_1836':
# /home/opam/.opam/4.05/.opam-switch/build/unison.2.51.4~rc3/_build/default/src/uutil.ml:39: undefined reference to `unsn_hash_univ_param'
# /usr/bin/ld: src/unison_lib.a(case.o): in function `camlCase__fun_1860':
# /home/opam/.opam/4.05/.opam-switch/build/unison.2.51.4~rc3/_build/default/src/uutil.ml:39: undefined reference to `unsn_hash_univ_param'
# /usr/bin/ld: src/unison_lib.a(case.o):/home/opam/.opam/4.05/.opam-switch/build/unison.2.51.4~rc3/_build/default/src/uutil.ml:39: more undefined references to `unsn_hash_univ_param' follow
# collect2: error: ld returned 1 exit status
# File "caml_startup", line 1:
# Error: Error during linking

-     ocamlopt src/linkgtk2.exe (exit 2)
- (cd _build/default && /home/opam/.opam/4.05/bin/ocamlopt.opt -w -40 -w -3-6-9-27-32-52 -g -o src/linkgtk2.exe /home/opam/.opam/4.05/lib/ocaml/unix.cmxa -I /home/opam/.opam/4.05/lib/ocaml /home/opam/.opam/4.05/lib/ocaml/threads/threads.cmxa -I /home/opam/.opam/4.05/lib/ocaml /home/opam/.opam/4.05/lib/ocaml/str.cmxa -I /home/opam/.opam/4.05/lib/ocaml src/lwt/lwt_lib.cmxa /home/opam/.opam/4.05/lib/ocaml/bigarray.cmxa -I /home/opam/.opam/4.05/lib/ocaml src/unison_lib.cmxa -I src /home/opam/.opam/4.05/lib/lablgtk2/lablgtk.cmxa /home/opam/.opam/4.05/lib/lablgtk2/gtkThread.cmx -I /home/opam/.opam/4.05/lib/lablgtk2 src/.linkgtk2.eobjs/native/uigtk2.cmx src/.linkgtk2.eobjs/native/linkgtk2.cmx)
- /usr/bin/ld: src/unison_lib.a(uutil.o): in function `camlUutil__hash_1212':
- /home/opam/.opam/4.05/.opam-switch/build/unison.2.51.4~rc3/_build/default/src/uutil.ml:39: undefined reference to `unsn_hash_univ_param'
- /usr/bin/ld: src/unison_lib.a(uutil.o): in function `camlUutil__43':
- :(.data+0x548): undefined reference to `unsn_hash_univ_param'
- /usr/bin/ld: src/unison_lib.a(case.o): in function `camlCase__fun_1815':
- /home/opam/.opam/4.05/.opam-switch/build/unison.2.51.4~rc3/_build/default/src/uutil.ml:39: undefined reference to `unsn_hash_univ_param'
- /usr/bin/ld: src/unison_lib.a(case.o): in function `camlCase__fun_1836':
- /home/opam/.opam/4.05/.opam-switch/build/unison.2.51.4~rc3/_build/default/src/uutil.ml:39: undefined reference to `unsn_hash_univ_param'
- /usr/bin/ld: src/unison_lib.a(case.o): in function `camlCase__fun_1860':
- /home/opam/.opam/4.05/.opam-switch/build/unison.2.51.4~rc3/_build/default/src/uutil.ml:39: undefined reference to `unsn_hash_univ_param'
- /usr/bin/ld: src/unison_lib.a(case.o):/home/opam/.opam/4.05/.opam-switch/build/unison.2.51.4~rc3/_build/default/src/uutil.ml:39: more undefined references to `unsn_hash_univ_param' follow
- collect2: error: ld returned 1 exit status
- File "caml_startup", line 1:
- Error: Error during linking
-     ocamlopt src/linktext.exe (exit 2)
- (cd _build/default && /home/opam/.opam/4.05/bin/ocamlopt.opt -w -40 -g -o src/linktext.exe /home/opam/.opam/4.05/lib/ocaml/str.cmxa -I /home/opam/.opam/4.05/lib/ocaml /home/opam/.opam/4.05/lib/ocaml/unix.cmxa -I /home/opam/.opam/4.05/lib/ocaml src/lwt/lwt_lib.cmxa /home/opam/.opam/4.05/lib/ocaml/bigarray.cmxa -I /home/opam/.opam/4.05/lib/ocaml src/unison_lib.cmxa -I src src/.linktext.eobjs/native/linktext.cmx)
- /usr/bin/ld: src/unison_lib.a(uutil.o): in function `camlUutil__hash_1212':
- /home/opam/.opam/4.05/.opam-switch/build/unison.2.51.4~rc3/_build/default/src/uutil.ml:39: undefined reference to `unsn_hash_univ_param'
- /usr/bin/ld: src/unison_lib.a(uutil.o): in function `camlUutil__43':
- :(.data+0x548): undefined reference to `unsn_hash_univ_param'
- /usr/bin/ld: src/unison_lib.a(case.o): in function `camlCase__fun_1815':
- /home/opam/.opam/4.05/.opam-switch/build/unison.2.51.4~rc3/_build/default/src/uutil.ml:39: undefined reference to `unsn_hash_univ_param'
- /usr/bin/ld: src/unison_lib.a(case.o): in function `camlCase__fun_1836':
- /home/opam/.opam/4.05/.opam-switch/build/unison.2.51.4~rc3/_build/default/src/uutil.ml:39: undefined reference to `unsn_hash_univ_param'
- /usr/bin/ld: src/unison_lib.a(case.o): in function `camlCase__fun_1860':
- /home/opam/.opam/4.05/.opam-switch/build/unison.2.51.4~rc3/_build/default/src/uutil.ml:39: undefined reference to `unsn_hash_univ_param'
- /usr/bin/ld: src/unison_lib.a(case.o):/home/opam/.opam/4.05/.opam-switch/build/unison.2.51.4~rc3/_build/default/src/uutil.ml:39: more undefined references to `unsn_hash_univ_param' follow
- collect2: error: ld returned 1 exit status
- File "caml_startup", line 1:
- Error: Error during linking
```

--- a/src/dune
+++ b/src/dune
@@ -7,7 +7,7 @@
         -w -3-6-9-10-26-27-32-34-35-38-39-50-52
         -warn-error -3-6-9-10-26-27-32-34-35-39-50-52
         -no-strict-sequence)
- (c_names bytearray_stubs osxsupport pty)
+ (c_names bytearray_stubs osxsupport pty hash_compat)
  (c_library_flags -lutil)
  (libraries str unix lwt_lib bigarray))
 

